core/src/com/unciv/ui/screens/diplomacyscreen/DiplomacyScreen.kt

```diff
@@ -791,8 +764,8 @@ class DiplomacyScreen(
         val tradeButton = "Trade".toTextButton()
         tradeButton.onClick {
             setTrade(otherCiv).apply {
-                tradeLogic.ourAvailableOffers.apply { remove(firstOrNull { it.type == TradeType.Treaty }) }
-                tradeLogic.theirAvailableOffers.apply { remove(firstOrNull { it.type == TradeType.Treaty }) }
+//                 tradeLogic.ourAvailableOffers.apply { remove(firstOrNull { it.type == TradeType.Treaty }) }
```

Comment:
What's this about?

Reply:
These two lines of code are very old, and I think they were meant to delete the research agreement button on the trade screen since it was offered through the diplomacy screen. Now that there is no longer a research button on the diplomacy screen, and we want the research button to show up, it is no longer needed.